mpd/__init__.py

```diff
@@ -21,120 +22,381 @@
-            port = int(item.conf['mpd_port'])
-        else:
-            port = 6600
-        host = item.conf['mpd_host']
-        lib.connection.Client.__init__(self, host, port, monitor=True)
-        self._sh = smarthome
-        self._cmd_lock = threading.Lock()
-        self._reply_lock = threading.Condition()
-        self._reply = {}
-        self._items = {}
-        self.terminator = b'\n'
-        self.found_terminator = self.handshake
-        for child in self._sh.find_children(item, 'mpd_listen'):
-            listen_to = child.conf['mpd_listen']
-            if listen_to in self._listen_keys:
-                self._items[listen_to] = child
-            if listen_to in self._current_keys:
-                self._items[self._current_keys[listen_to]] = child
-        for child in self._sh.find_children(item, 'mpd_send'):
-            send_to = child.conf['mpd_send']
-            if send_to in self._bool_keys:
-                child.add_method_trigger(self._send_bool)
-            elif send_to == 'volume':
-                child.add_method_trigger(self._send_volume)
-            elif send_to == 'value':
-                child.add_method_trigger(self._send_value)
-            else:
-                child.add_method_trigger(self._send_command)
-        for child in self._sh.find_children(item, 'mpd_file'):
-            child.add_method_trigger(self._play_file)
-        for child in self._sh.find_children(item, 'mpd_localplaylist'):
-            child.add_method_trigger(self._play_localplaylist)
-        # adding item methods
-        item.command = self.command
-        item.play_file = self.play_file
-        item.add_file = self.add_file
-
-    def command(self, command, wait=True):
-        return self._send(command, wait)
-
-    def play_file(self, url):
-        play = self._parse_url(url)
-        if play == []:
+    def update_statusitems(self,warn):
+        if not self.connected:
+            if warn:
+                self.loggercmd("update_status while not connected",'e')
             return
-        self._send('clear', False)
-        for url in play:
-            self._send("add {0}".format(url), False)
-        self._send('play', False)
+        if (len(self._status_items) <= 0):
+            if warn:
+                self.loggercmd("status: no items to refresh",'w')
+            return
+        self.loggercmd("requesting status",'d')
+        status = self._send('status')
+        self.refreshItems(self._status_items,status,warn)
 
-    def play_localplaylist(self, file):
-        self._send('clear', False)
-        self._send('load ' + file, False)
-        self._send('play', False)
+    def update_currentsong(self,warn):
+        if not self.connected:
+            if warn:
+                self.loggercmd("update_currentsong while not connected",'e')
+            return
+        if (len(self._currentsong_items) <= 0):
+            if warn:
+                self.loggercmd("currentsong: no items to refresh",'e')
```

Comment:
Shouldn't it be just a warning here?

Reply:
Yes I can see them and of course you are right. I have to be blind... I was sure they are warnings. Will harmonize them to warnings with the next commit.